backend/controllers/order.js

```diff
@@ -0,0 +1,106 @@
+const ReasonPhrases = require('http-status-codes').ReasonPhrases;
+const StatusCodes = require('http-status-codes').StatusCodes;
+const Order = require('../models/order');
+
+// Handle index actions
+exports.index = function (req, res) {
+  Order.find(function (err, order) {
+    if (err) {
+      console.log(err);
+      res.status(StatusCodes.INTERNAL_SERVER_ERROR).json({
+        message: ReasonPhrases.INTERNAL_SERVER_ERROR,
+      });
+    } else {
+      res.status(StatusCodes.OK).json(order.map(order => {
+        return order;
+      }));
+    }
+  });
+};
+
+// Handle create order actions
+exports.new = function (req, res) {
+  const order = new Order(
+    {
+      product_id: req.body.product_id,
+      buyer_id: req.body.buyer_id,
+      address: {
+        street: req.body.address.street,
+        street_number: req.body.address.street_number,
+        city: req.body.address.city,
+        postal_code: req.body.address.postal_code,
+      },
+      state: req.body.type,
+      timestamp: {
+        created: req.body.timestamp.created
+      }
+    }
+  );
+
+  order.save(function (err, order) {
+    if (err) {
+      console.log(err);
+      res.status(StatusCodes.BAD_REQUEST).json({
+        message: ReasonPhrases.BAD_REQUEST,
+      });
+    } else {
+      res.status(StatusCodes.CREATED).json(order.toJSON());
+    }
+  });
+};
+
+// Handle view order info
+exports.view = function (req, res) {
+  Order.findById(req.params.order_id, function (err, order) {
+    if (err) {
+      console.log(err);
+      res.status(StatusCodes.NOT_FOUND).json({
+        message: ReasonPhrases.NOT_FOUND,
+      });
+    } else {
+      res.status(StatusCodes.OK).json(order.toJSON());
+    }
+  });
+};
+
+// Handle update order info
+exports.update = function (req, res) {
+  Order.findById(req.params.order_id, function (err, order) {
+    if (err) {
+      console.log(err);
+      res.status(StatusCodes.NOT_FOUND).json({
+        message: ReasonPhrases.NOT_FOUND,
+      });
+    } else {
+      order.state = req.body.state ? req.body.state : order.state;
```

Comment:
Maybe we should add some regulations here so that the state can't be changed in every direction. But not sure if that makes sense to do. What do you think about that?

Reply:
Hmm I think it does. Especially once an order is cancelled it shouldn't be possible to simply 'reactivate' it